BasicSample/Assets/Shared/Scripts/MixedRealityProjectSelectionWindow.cs

```diff
@@ -0,0 +1,138 @@
+using UnityEngine;
+using UnityEngine.XR.Management;
+using UnityEngine.XR.OpenXR;
+using UnityEngine.XR.OpenXR.Features;
+
+
+namespace Microsoft.MixedReality.OpenXR.BasicSample
+{
+    enum MixedRealityProjectConfiguration
+    {
+        None,
+        RunNativelyonPCVR,
+        RunNativelyonHL2,
+        RunRemotelyonUWP,
+        RunRemotelyonWin32
+    }
+    public class MixedRealityProjectSelectionWindow : EditorWindow
+    {
+        private MixedRealityProjectConfiguration m_selectedMRConfiguration;
+        private const float Default_Window_Height = 300.0f;
+        private const float Default_Window_Width = 400.0f;
+
+        public static MixedRealityProjectSelectionWindow Instance { get; private set; }
+        public static bool IsOpen => Instance != null;
+        private static bool DisablePopup = false;
+        private static bool ShowfromMenu = false;
+
+        [MenuItem("Mixed Reality/Quick Setup", false, 499)]
+        private static void ShowWindowFromMenu()
+        {
+            ShowfromMenu = true;
+            ShowWindow();
+        }
+
+        public static void ShowWindow()
+        {
+            // There should be only one configurator window open as a "pop-up". If already open, then just force focus on our instance
+            if (IsOpen)
+            {
+                Instance.Focus();
+            }
+            else
+            {
+                if(!DisablePopup || ShowfromMenu)
+                {
+                    var window = CreateInstance<MixedRealityProjectSelectionWindow>();
+                    window.titleContent = new GUIContent("MixedReality Project Selection Window", EditorGUIUtility.IconContent("_Popup").image);
+                    window.position = new Rect(Screen.width / 2.0f, Screen.height / 2.0f, Default_Window_Height, Default_Window_Width);
+                    window.ShowUtility();
+                }
+
+            }
+        }
+
+        private void OnGUI()
+        {
+            EditorGUIUtility.labelWidth = 400;
+            GUILayout.TextField("Change the Unity project settings for Mixed Reality Scenario",GUILayout.Width(350));
+            m_selectedMRConfiguration = GUILayout.Button("Win32 app running on PC VR") ? MixedRealityProjectConfiguration.RunNativelyonPCVR :
+                                        GUILayout.Button("UWP app running on HoloLens 2") ? MixedRealityProjectConfiguration.RunNativelyonHL2 :
+                                        GUILayout.Button("Holographic Remoting remote UWP app") ? MixedRealityProjectConfiguration.RunRemotelyonUWP :
+                                        GUILayout.Button("Holographic Remoting remote Win32 app") ? MixedRealityProjectConfiguration.RunRemotelyonWin32 :
+                                        MixedRealityProjectConfiguration.None;
+            DisablePopup = GUILayout.Toggle(DisablePopup, "Don't show up this popup anymore");
+            ApplySelectedConfiguration(m_selectedMRConfiguration);
```

Comment:
If no button is pressed, m_selectedMRConfiguration will be MixedRealityProjectConfiguration.None - in that case, should we just not call this ApplySelectedConfiguration method? It seems like currently there could be some side effects of passing None into that method and running it each OnGUI().

Reply:
not sure when you tried this last, but with the latest change there is no "Apply" button anymore. If the user does not want to choose any setting, he simply can close the gui
https://github.com/microsoft/OpenXR-Unity-MixedReality-Samples/pull/69#issue-1192154537

Reply:
can you elaborate the side effect? I see that it just returns.

Reply:
You're right - this is an old comment from a different approach. Cleaning it up